client.go

```diff
@@ -1150,7 +1155,7 @@ func (c *Client[TTx]) ID() string {
 	return c.config.ID
 }
 
-func insertParamsFromConfigArgsAndOptions(archetype *baseservice.Archetype, config *Config, args JobArgs, insertOpts *InsertOpts) (*riverdriver.JobInsertFastParams, *dbunique.UniqueOpts, error) {
```

Comment:
So not 100% sure on this one yet, but the "insert" part of the middleware needs to receive a type that's _not_ a `JobRow` because we don't have a job row yet. I basically took `JobInsertParams`, duplicated it, and promoted it to `rivertype`. The types are different for now, but they can be type converted to one another because they have the same fields. I basically did this because I like the naming of `JobInsertParams`, but they could also be the same type or even slightly different types with a couple fields dropped (`CreatedAt` for example, which is only needed for time injection).

Either way, `JobInsertParams` stays internal for now, so it should leave refactoring possibility ...